case-study.Rmd

```diff
@@ -0,0 +1,489 @@
+    stop("Detect more than one GSEA result folder. Please remove the unwanted
+         result folder and only keep one.")
+  }
+} else {
+  dir.create(GSEA_folder)
+  run_GSEA(GSEA_file, GSEA_folder, GSEA_output_name, data_raw, data_pheno,
+             KEGG_subset)
+}
+
+# read in GSEA results
+gsea_con1 <- readr::read_tsv(Sys.glob(
+  file.path(GSEA_folder, paste0(GSEA_output_name, ".Gsea*"),
+            "gsea_report_for_con1_*.xls")), col_names = TRUE)
+gsea_con2 <- readr::read_tsv(Sys.glob(
+  file.path(GSEA_folder, paste0(GSEA_output_name, ".Gsea*"),
+            "gsea_report_for_con2_*.xls")), col_names = TRUE)
+
+# get significant results
+gsea_con1_sig <- gsea_con1[gsea_con1$`FDR q-val` <= 0.05, ]
+gsea_con2_sig <- gsea_con2[gsea_con2$`FDR q-val` <= 0.05, ]
+KEGG_GSEA <- unique(c(gsea_con1_sig$NAME, gsea_con2_sig$NAME))
+
+```
+
+Now let's compare the significant KEGG pathways in GSEA with the KEGG pathways
+associated with the active signatures.
+
+```{r}
+# GSEA automatically converts pathway names to upper case, so we also convert
+# the pathways identified by ADAGEpath.
+KEGG_ADAGE <- toupper(as.character(unique(pathway_association$geneset)))
+
+# make a venn diagram for pathway overlap
+grid.newpage()
+invisible(draw.pairwise.venn(length(KEGG_GSEA), length(KEGG_ADAGE),
+                             length(intersect(KEGG_GSEA, KEGG_ADAGE)),
+                             c("GSEA", "ADAGE"), fill = c("red", "blue")))
+```
+
+As we can see, GSEA and ADAGEpath agree on many pathways. Here we look into
+pathways significantly enriched in GSEA but not associated with chosen active
+signatures. We check whether they are assoicated with any ADAGE signatures and
+how differentially active the associated signatures are.
+
+```{r}
+# built a match table between normal case and upper case
+match_table <- data.frame(normal_case = names(KEGG_subset),
+                          upper_case = toupper(names(KEGG_subset)),
+                          stringsAsFactors = FALSE)
+
+# pathways enriched in GSEA but not associated with ADAGE signatures
+# in the top 10 layers of pareto fronts
+GSEA_only <- match_table$normal_case[match(setdiff(KEGG_GSEA, KEGG_ADAGE),
+                                           match_table$upper_case)]
+
+# check whether GSEA-only KEGG pathways are associated with any signatures
+# this step will take a while because it checks association between all signatures
+# and all pathways
+associated_sigs <- find_associated_signatures(input_genesets = GSEA_only,
+                                              model = model,
+                                              gene_set_list = KEGG_subset,
+                                              significance_cutoff = 0.05,
+                                              signature_limma_result = limma_result)
+associated_sigs <- unlist(associated_sigs)
+
+# see how these signatures perform in the activation test
+plot_volcano(limma_result, highlight_signatures = associated_sigs,
```

Comment:
I decided to display both differentially active signatures picked previously and these associated signatures together in the volcano plot. In this way, people can quickly notice that some of the associated signatures are close to meet the activation cutoff.